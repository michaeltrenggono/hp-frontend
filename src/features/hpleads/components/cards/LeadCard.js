import React, {useCallback} from "react";
import {useDispatch} from "react-redux";
import styles from "./LeadCard.module.scss";
import {LeadCardRow} from "./LeadCardRow";
import {IconInitial} from "../IconInitial";
import {ButtonStrip} from "../ButtonStrip";
import {updateLeadWithIdAsync} from "../../leadsSlice";
import {formatPrice, formatDateTime} from "../../../../helpers";

export const LeadCard = ({className, lead}) => {

    const dispatch = useDispatch();

    const updateLead = useCallback((status) => {
        dispatch(updateLeadWithIdAsync({id: lead.id, status}));
    }, [dispatch, lead.id]);

    return (
        <div className={`${styles.leadCardContainer} ${className}`}>
            <div className="card">
                <div className="card-body p-0">
                    <LeadCardRow>
                        <div className={styles.header}>
                            <IconInitial initial={lead.customer.first_name.charAt(0)}/>
                            <div>
                                <div className="font-weight-bold text-dark">{lead.customer.first_name}</div>
                                <div>{formatDateTime(lead.updated_at !== null ? lead.updated_at : lead.created_at)}</div>
                            </div>
                        </div>
                    </LeadCardRow>

                    <LeadCardRow>
                        <div className={styles.summary}>
                            <span>
                                <i className="fa fa-map-marker-alt"/> {lead.suburb}
                            </span>
                            <span>
                                <i className="fa fa-briefcase"/> {lead.category}
                            </span>
                            <span>
                                Job ID: {lead.id}
                            </span>
                            {lead.status === "Accepted" && (
                                <span>
                                    {formatPrice(lead.price)} Lead Invitation
                                </span>
                            )}
                        </div>
                    </LeadCardRow>

                    <LeadCardRow>
                        {lead.status === "Accepted" && (
                            <div className={styles.customerContact}>
                                <span>
                                    <i className="fa fa-phone-alt"/> <a href={`tel:${lead.customer.phone}`} className="text-orange font-weight-bold">{lead.customer.phone}</a>
                                </span>
                                <span>
                                    <i className="fa fa-envelope"/> <a href={`mailto:${lead.customer.email}`} className="text-orange font-weight-bold">{lead.customer.email}</a>
                                </span>
                            </div>
                        )}
                        {lead.description}
                    </LeadCardRow>

                    <LeadCardRow className="d-md-none">
                        <strong>{formatPrice(lead.price)}</strong> Lead Invitation
                    </LeadCardRow>

                    {lead.status === "Invited" && (
                        <LeadCardRow>
                            <div className={`d-flex align-items-center cta ${styles.cta}`}>
                                <ButtonStrip label="Accept" className={styles.accept} onClick={() => updateLead("Accepted")}/>
                                <ButtonStrip label="Decline" className={styles.decline} onClick={() => updateLead("Declined")}/>
                                <span className="d-none d-md-inline">
                                    <strong>{formatPrice(lead.price)}</strong> Lead Invitation
                                </span>
                            </div>
                        </LeadCardRow>
                    )}
                </div>
            </div>
        </div>
    );
};
