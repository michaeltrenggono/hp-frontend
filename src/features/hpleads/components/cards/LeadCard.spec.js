import React from 'react';
import { shallow } from 'enzyme';
import {LeadCard} from './LeadCard';
import { Provider } from "react-redux";

const ReduxProvider = ({ children, reduxStore }) => (
    <Provider store={reduxStore}>{children}</Provider>
);

describe('Should render leadCard and match snapshot', () => {
    it('should render LeadCard correctly', () => {
        const card = shallow(
            <ReduxProvider>
                <LeadCard lead={{id:1, status: "Invited"}}/>
            </ReduxProvider>
        );

        expect(card).toMatchSnapshot();
    });

    it('should render LeadCard correctly when Accepted', () => {
        const card = shallow(
            <ReduxProvider>
                <LeadCard lead={{id:2, status: "Accepted"}}/>
            </ReduxProvider>
        );

        expect(card).toMatchSnapshot();
    });
});
