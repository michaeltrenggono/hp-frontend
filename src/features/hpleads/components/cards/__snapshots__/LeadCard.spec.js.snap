// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render leadCard and match snapshot should render LeadCard correctly 1`] = `ShallowWrapper {}`;

exports[`Should render leadCard and match snapshot should render LeadCard correctly when Accepted 1`] = `ShallowWrapper {}`;
